fahrschein-http-apache/src/main/java/org/zalando/fahrschein/http/apache/HttpComponentsRequest.java

```diff
@@ -55,7 +55,11 @@ public String getMethod() {
 
     @Override
     public URI getURI() {
-        return this.httpRequest.getURI();
+        try {
+            return this.httpRequest.getUri();
+        } catch (URISyntaxException e) {
```

Comment:
This one is mean – throwing a syntax exception on the getter, and not on the setter. 😓

Reply:
Should we file a Technical Debt ticket for it and handle it in a separate PR?

Reply:
No, I don't think there's anything we can do here. The URISyntaxException is coming from Apache, and I really don't want to expose it like this in our own API definition.

Reply:
.. we could eliminate the problem via `URI.create(this.httpRequest.getRequestUri());`